src/parser/data-purge-module.ts

```diff
@@ -0,0 +1,38 @@
+import configuration from "../configuration/config-reader";
+import { GetActivity } from "../get-activity";
+import { Module, Result } from "./processor";
+
+/**
+ * Removes the data in the comments that we do not want to be processed.
+ */
+export class DataPurgeModule implements Module {
+  readonly configuration = configuration.dataPurge;
+
+  get enabled(): boolean {
+    return this.configuration.enabled;
+  }
+
+  transform(data: Readonly<GetActivity>, result: Result) {
+    for (const value of data.allComments) {
+      if (value.body && value.user?.login && result[value.user.login]) {
+        const newContent = value.body
+          .replace(/^>.*$/gm, "")
+          .replace(/[\r\n]+/g, " ")
+          .replace(/\[.*?\]\(.*?\)/g, "")
+          .replace(/^\/\S+/g, "")
+          .trim();
+        if (newContent.length) {
+          result[value.user.login].comments = [
+            ...(result[value.user.login].comments ?? []),
```

Comment:
Since only comments that got transformed get added to the array, no. At that point the array comes empty and gets filled after having the original comment purged.